modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbarLookup.ts

```diff
@@ -31,18 +75,22 @@ const lookup = (scopes: ScopedToolbars, editor: Editor): Option<LookupResult> =>
     return Option.none();
   }
 
-  return matchTargetWith(startNode, scopes.inNodeScope).orThunk(() => {
-    return matchTargetWith(startNode, scopes.inEditorScope).orThunk(() => {
+  return matchStartNode(startNode, scopes.inNodeScope, scopes.inEditorScope).orThunk(() => {
+    return TransformFind.ancestor(startNode, (ancestorElem) => {
       // Don't continue to traverse if the start node is the root node
-      if (isRoot(startNode)) {
+      if (isRoot(ancestorElem)) {
```

Comment:
This isn't correct, as this should be before it tries to walk up the tree. Otherwise if the start node is the root, it'll walk outside the root node.

Reply:
Probably a bad refactor, will fix